Assets/MRTK/SDK/Features/Utilities/Solvers/HandConstraintPalmUp.cs

```diff
@@ -163,5 +197,125 @@ protected override bool IsValidController(IMixedRealityController controller)
 
             return true;
         }
+
+        /// <summary>
+        /// Checks to see if the palm is currently facing the user
+        /// </summary>
+        /// <param name="jointedHand"></param>
+        /// <param name="palmPose"></param>
+        /// <param name="palmCameraAngle"></param>
+        /// <returns></returns>
+        private bool IsPalmFacingUpwards(IMixedRealityHand jointedHand, MixedRealityPose palmPose, float palmCameraAngle)
+        {
+            if (requireFlatHand)
+            {
+                // Check if the triangle's normal formed from the palm, to index, to ring finger tip roughly matches the palm normal.
+                MixedRealityPose indexTipPose, ringTipPose;
+
+                if (jointedHand.TryGetJoint(TrackedHandJoint.IndexTip, out indexTipPose) &&
+                    jointedHand.TryGetJoint(TrackedHandJoint.RingTip, out ringTipPose))
+                {
+                    var handNormal = Vector3.Cross(indexTipPose.Position - palmPose.Position,
+                                                   ringTipPose.Position - indexTipPose.Position).normalized;
+                    handNormal *= (jointedHand.ControllerHandedness == Handedness.Right) ? 1.0f : -1.0f;
+
+                    if (Vector3.Angle(palmPose.Up, handNormal) > flatHandThreshold)
+                    {
+                        return false;
+                    }
+                }
+            }
+
+            // Check if the palm angle meets the prescribed threshold
+            return palmCameraAngle < facingCameraTrackingThreshold;
+        }
+
+        /// <summary>
+        /// Checks to see if the user is currently gazing at the activation point; it first attempts to do so 
+        /// using eyegaze, and then falls back to head-based gaze if eyegaze isn't available for use.
+        /// </summary>
+        /// <param name="jointedHand"></param>
+        /// <returns></returns>
+        private bool IsUserGazingAtActivationPoint(IMixedRealityHand jointedHand)
```

Comment:
I'm not clear on what the activation point is? Can you clarify this?

Reply:
Vector3 activationPoint = Vector3.Lerp(pinkyKnuckle.Position, wrist.Position, .5f);
That's what it is - the outer edge of the hand. Why use this? It seems like this should change depending on what hand position you're targeting. [Ulnar vs Radial](https://github.com/microsoft/MixedRealityToolkit-Unity/blob/mrtk_release/Documentation/README_Solver.md#hand-menu-with-handconstraint-and-handconstraintpalmup).

Reply:
Sorry, you caught this pretty early on (we were iterating in @cre8ivepark's side branch for that reason); I definitely forgot to make this respect the various hand positions available to the user. Will implement that later in the week.

An activation point is a reference point on the generated hand plane that is used in conjunction with distance calculations to determine whether the user's gaze is currently within range (based on the threshold) to 'activate' the menu (return true for IsValidController for the initial activation of the menu, or reattachment of the menu).